execution_engine/src/tracking_copy/mod.rs

```diff
@@ -342,6 +345,24 @@ impl<R: StateReader<Key, StoredValue>> TrackingCopy<R> {
         self.effects.push(transform);
     }
 
+    /// Caches the emitted message and writes the message topic summary under the specified key.
+    ///
+    /// This function does not check the types for the key and the value so the caller should
+    /// correctly set the type. The `message_topic_key` should be of the `Key::MessageTopic`
+    /// variant and the `message_topic_summary` should be of the `StoredValue::Message` variant.
+    pub fn emit_message(
+        &mut self,
+        message_topic_key: Key,
+        message_topic_summary: StoredValue,
+        message_key: Key,
+        message_value: StoredValue,
+        message: Message,
```

Comment:
I'd have a similar comment here to what I mentioned above, I think we could pass in more explicitly typed arguments here, instead of the enums

Reply:
Yeah, makes sense, though I think the tracking copy should probably only deal with `StoredValue`s rather than the inner objects to keep it more abstract.